File: app/src/main/java/igrek/todotree/model/treeitem/RootTreeItem.java
package igrek.todotree.model.treeitem;


public class RootTreeItem extends AbstractTreeItem {
	
	public RootTreeItem() {
		super(null);
	}
	
	@Override
	public RootTreeItem clone() {
		return new RootTreeItem().copyChildren(this);
	}
	
	@Override
	public String getDisplayName() {
		return "/";
	}
	
	@Override
	public String getTypeName() {
		return "/";
	}
	
	@Override
	public AbstractTreeItem getParent() {
		return null;
	}
}

File: app/src/main/java/igrek/todotree/domain/treeitem/RootTreeItem.kt
package igrek.todotree.domain.treeitem

class RootTreeItem : AbstractTreeItem(null) {
    override fun clone(): RootTreeItem {
        return RootTreeItem().copyChildren(this)
    }

    override val displayName: String
        get() = "/"

    override val typeName: String
        get() = "/"

    override fun getParent(): AbstractTreeItem? {
        return null
    }
}